api/src/main/java/io/opentelemetry/trace/spi/TraceProvider.java

```diff
@@ -30,7 +30,7 @@
  * @see io.opentelemetry.OpenTelemetry
  */
 @ThreadSafe
-public interface TracerRegistryProvider {
+public interface TraceProvider {
```

Comment:
How about `TracerProvider`, since this provides tracers, not traces?

Reply:
I was thinking to use the name of the package with the idea that it provides the implementation of the whole package. Maybe we rename the whole package to `traces` (separate PR maybe)?

Reply:
how about "tracing", then? And, yes, a follow-up PR would be fine.
So, we have the package `io.opentelemetry.tracing...` and `TracingProvider`.  I like that much more.

Reply:
*If* we happen to rename things, I'd be up for `tracing` rather than `traces` (even though we have `metrics`, as in a plural noun ;) )

That being said, having this as either `TracerProvider` or `TraceProvider` is fine IHMO.